### PR TITLE
Show active viewport dimensions & button to rotate

### PR DESCRIPTION
**Pull request by lucas-carl**
Issue: #5879

## What I did
- Show active pixel dimensions in viewport widget when a viewport is selected
- Added button to flip the viewport
- Show title of active view in button
- Added functionality to reset the viewport by double-clicking the button

### CODE REVIEW

File: addons/viewport/src/Tool.js [hunk 1]
```diff
@@ -118,11 +136,24 @@ export default class ViewportTool extends Component {
...
     const item = items.find(i => i.id === selected);
 
+    let viewportX = 0;
+    let viewportY = 0;
+    if (item) {
+      const height = item.value.height.replace('px', '');
+      const width = item.value.width.replace('px', '');
+
+      viewportX = isRotated ? height : width;
+      viewportY = isRotated ? width : height;
```
Comment (backbone87) [inline addons/viewport/src/Tool.js]: why is this logic here? there must already exist rotation logic, from the dropdown menu?

**Comment by backbone87** [comment]
If i click the rotate button, the button keeps being highlighted until i click somewhere else.

Additional feature wishes:
- Display size when no viewport is selected (basically the iframes body size)
- Display the name of the selected view (if any)
- Append P (Portrait) or L (Landscape) to the viewport name

**Review by ndelangen — APPROVED** [review]

**Comment by shilman** [comment]
This is awesome. Could use some visual treatment (cc @domyen) but we can do that before the 5.1 release. Great work!

**Review by shilman — APPROVED** [review]

**Comment by lucas-carl** [comment]
@shilman @backbone87 I will try to implement these suggestions.
Also, I had the idea that you can double-click on the active viewport's name to reset the viewport.

**Comment by backbone87** [comment]
sadly
> Display size when no viewport is selected (basically the iframes body size)

is missing

**Pull request merged**